How to join a working group (#2158)
graphql/graphql.github.io: src/pages/community/contribute/project-meetings.mdx

import { Cards } from "../../../components/cards"
import { Callout } from 'nextra/components'
import { Steps } from 'nextra/components'
 
# Project Meetings

> All GraphQL technical meetings are open to anyone who is interested, so long as they first complete the [GraphQL Specification Membership agreement](https://github.com/graphql/graphql-wg/tree/HEAD/membership). Participation in the technical meetings is free. 

Reviewer: ```suggestion
All GraphQL technical meetings are free and open to anyone who is interested, so long as they first complete the [GraphQL Specification Membership agreement](https://github.com/graphql/graphql-wg/tree/HEAD/membership).
```

All GraphQL technical meetings should appear on the [GraphQL public calendar](https://calendar.graphql.org). If a project's meeting isn't on the calendar or is incorrect, please email operations@graphql.org.

## Who can join GraphQL technical meetings?

Anyone who has signed the free [GraphQL Specification Membership agreement](https://github.com/graphql/graphql-wg/tree/HEAD/membership).

## Quick guide: How to join 

1. Sign the GraphQL Specification Membership agreement (first time only)
    1. Open a pull request against the [EasyCLA repository](https://github.com/graphql/easycla)
    1. Follow the steps provided by the EasyCLA bot
    1. Choose to sign the CLA as an individual or on behalf of a corporation
1. Open a pull request against the meeting's agenda (forking if necessary)
    1. Add your name and affiliation to the agenda
    1. Add any agenda items, with a time estimate (optional)
1. Join using the provided Zoom link
1. Notes are taken by AI, and volunteers are welcome to add to a shared notes doc during the meeting
1. Meetings are recorded and later made available on the [GraphQL Foundation Working Groups YouTube channel](https://www.youtube.com/@GraphQLFoundation)

> [See below for detailed instructions](#how-to-join-a-meeting) on how to sign the membership agreement and join a meeting using a GitHub pull request.

Reviewer: ```suggestion
[See below for detailed instructions](#how-to-join-a-meeting) on how to sign the membership agreement and join a meeting using a GitHub pull request; here's a brief summary:

1. Open a pull request against the meeting's agenda (forking if necessary)
    1. Add your name and affiliation to the agenda
    1. Add any agenda items, with a time estimate (optional)
    1. First time? Sign the CLA (as an individual or on behalf of an organization) as instructed by the EasyCLA bot
1. Join using the provided Zoom link
1. Notes are taken by AI, and volunteers are encouraged to add to a shared notes doc during the meeting; these will be filed in the `notes` folder
1. Meetings are recorded and later made available on the [GraphQL Foundation Working Groups YouTube channel](https://www.youtube.com/@GraphQLFoundation)
```

      description: "A monthly virtual meeting of maintainers of commonly used GraphQL libraries and tools and significant contributors in the Javascript ecosystem, operated by the GraphQL Foundation.",
    },
    {
      title: "GraphQL/OTel working group",
      link: "https://github.com/graphql/otel-wg",
      description: "Defining the best practices and semantic conventions for using OpenTelemetry with GraphQL.",
    },
    {
      title: "GraphQL-over-HTTP working group",
      link: "https://github.com/graphql/graphql-over-http/tree/main/working-group",
      description: "Working on a specification that allows GraphQL clients and servers with different implementations and technology stacks to interact freely over HTTP.",
    },
    {
      title: "Nullability working group",
      link: "https://github.com/graphql/nullability-wg",
      description: "Giving clients a way to influence or override nullability checks within queries.",
    }
  ]}
  />

## How to join a meeting

If you have questions about this process, please ask [operations@graphql.org](mailto:operations@graphql.org).
 
<Steps>
### Make a pull request to the EasyCLA repository
 

> The GraphQL Specification Membership agreement is managed by a neutral GitHub repository where you can open a pull request and initiate the EasyCLA signing process. All pull requests will be closed periodically. You only need to do this one time, and it can be signed by individual contributors or their employers.

Reviewer: ```suggestion
To see if you have a GraphQL Specification Membership agreement in place you can open a pull request against the [EasyCLA GitHub repository](https://github.com/graphql/easycla) to initiate the EasyCLA signing process; if the bot responds with green "CLA signed" then you already have a CLA in place. The CLA only needs to be signed one time, and it can be signed by individual contributors or their employers. All pull requests will be closed periodically.
```

> You can read about the full process, including details about whether to sign as an individual or under an organization, [at the EasyCLA GitHub repository](https://github.com/graphql/graphql-wg/tree/main/membership). 

Reviewer: ```suggestion
You can read about the agreement, including details about whether to sign as an individual or under an organization, [in the specification membership details page](https://github.com/graphql/graphql-wg/tree/main/membership). 
```

> 1. Begin by pressing the edit button in the top right corner of the "README" file

Reviewer: ```suggestion
1. Begin by pressing the [edit button](https://github.com/graphql/EasyCLA/edit/main/README.md) in the top right corner of the ["README" file](https://github.com/graphql/EasyCLA/blob/main/README.md)
```

2. Press the Edit button in the top right corner of the agenda file, and when prompted create a fork by pressing the "Fork this repository" button
3. The file is written in markdown and is generally easy to read. Scroll down to `## Attendees`. Add your information following the format of the other attendees in the list. You can preview your change in a rich text format by pressing "Preview" in the top left of the editing pane

![Create pull request to the agenda](./images/development/Step3-3.png)

4. If applicable, add an agenda item under the `## Agenda` heading, make sure you add your item to the end of the list and add an estimated duration of discussion
5. Press "Commit changes", fill in the details of your changes in the popup, and then press "Propose changes"
6. Now you can review your changes and create your pull request to the agenda by pressing "Create pull request"
7. If you successfully signed the EasyCLA, you will pass the Easy CLA check and your pull request may be automatically merged if you pass the automatic checks. 

> 8. If you don't pass the automatic checks, don't worry, reach out on the [community Discord](https://discord.graphql.org/) or request a [TSC member](https://github.com/graphql/graphql-wg/blob/HEAD/GraphQL-TSC.md#tsc-membership) as a reviewer of your pull request. 

Reviewer: ```suggestion
8. If you successfully signed the EasyCLA but your PR is not automatically merged, don't worry; pull requests will be reviewed and merged before or at the start of the meeting.
9. If you have issues or questions about this process, please ask [operations@graphql.org](mailto:operations@graphql.org).
```
